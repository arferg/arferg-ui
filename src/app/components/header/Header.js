import React, {useEffect} from "react";
import "./header.scss"
import Option from "./Option";
import {ToastProvider} from "react-toast-notifications";
import {getHeaderDetails} from "../../store/header/Actions";
import useGetStarWarsPerson from "../../hooks/useGetStarWarsPerson";


const Header = props => {
    const {
        options
    } = props

    const {data, isError, isLoading} = useGetStarWarsPerson("123")

    useEffect(async () => {
        getHeaderDetails()
    }, [])

    if (isLoading) return "Loading...";

    if (isError) return "An error has occurred: " + error.message;

    return (
        <ToastProvider>
            <div className={"header"}>
                logo

                {options.map((option, index) =>
                    <Option key={index}
                            details={option}
                    />)}
            </div>
        </ToastProvider>
    )
}

export default Header