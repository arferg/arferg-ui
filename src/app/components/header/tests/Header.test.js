import React from "react";
import "@testing-library/jest-dom"
import {render} from "@testing-library/react"
import Header from "../Header"

jest.mock('../Option', () => () => <div/>)

const props = {
    options: []
};

describe("Header Tests", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    describe("Should render ", () => {
        it("correctly with the product name", () => {
            const {asFragment} = render(<Header {...props}/>)
            expect(asFragment()).toMatchSnapshot()
        });
    })
})
