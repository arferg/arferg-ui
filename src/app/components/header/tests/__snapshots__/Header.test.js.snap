// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Should render  correctly with the product name 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    logo
  </div>
</DocumentFragment>
`;
